builder/vmware/common/step_vnc_connect.go

```diff
@@ -0,0 +1,137 @@
+package common
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/url"
+
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+	"github.com/mitchellh/go-vnc"
+	"golang.org/x/net/websocket"
+)
+
+type StepVNCConnect struct {
+	VNCEnabled       bool
+	VNCOverWebsocket bool
+	DriverConfig     *DriverConfig
+}
+
+func (s *StepVNCConnect) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	if !s.VNCOverWebsocket && !s.VNCEnabled {
+		return multistep.ActionContinue
+	}
+	ui := state.Get("ui").(packer.Ui)
+
+	var c *vnc.ClientConn
+	var err error
+
+	if s.VNCOverWebsocket {
+		ui.Say("Connecting to VNC over websocket...")
+		c, err = s.ConnectVNCOverWebsocketClient(state)
+	} else {
+		ui.Say("Connecting to VNC...")
+		c, err = s.ConnectVNC(state)
+	}
+
+	if err != nil {
+		state.Put("error", err)
+		ui.Error(err.Error())
+		return multistep.ActionHalt
+	}
+
+	state.Put("vnc_conn", c)
+	return multistep.ActionContinue
+}
+
+func (s *StepVNCConnect) ConnectVNCOverWebsocketClient(state multistep.StateBag) (*vnc.ClientConn, error) {
+	driver := state.Get("driver").(*ESX5Driver)
+
+	// Acquire websocket ticket
+	ticket, err := driver.AcquireVNCOverWebsocketTicket()
+	if err != nil {
+		err := fmt.Errorf("Error acquiring vnc over websocket ticket: %s", err)
+		state.Put("error", err)
+		return nil, err
+	}
+	host := ticket.Host
+	if len(host) == 0 {
+		host = s.DriverConfig.RemoteHost
+	}
+	port := ticket.Port
+	if port == 0 {
+		port = 443
+	}
+
+	u, err := url.Parse(fmt.Sprintf("wss://%s:%d/ticket/%s", host, port, ticket.Ticket))
+	if err != nil {
+		err := fmt.Errorf("Error parsing websocket url: %s\n", err)
+		state.Put("error", err)
+		return nil, err
+	}
+	origin, err := url.Parse("http://localhost")
+	if err != nil {
+		err := fmt.Errorf("parse websocket origin url: %s\n", err)
+		state.Put("error", err)
+		return nil, err
+	}
+
+	// Create the websocket connection and set it to a BinaryFrame
+	websocketConfig := &websocket.Config{
+		Location:  u,
+		Origin:    origin,
+		TlsConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
Should this be set to the main `insecure_connection` setting rather than hardcode to true?

Reply:
Good idea! I'll do that.